ebs_snapshots/snapshot_manager.py

```diff
@@ -119,7 +186,87 @@ def _ensure_snapshot(connection, volume, interval, name):
     elif interval == 'yearly' and min_delta > 3600*24*365:
         _create_snapshot(connection, volume, name)
     else:
-        logging.info(kayvee.formatLog("ebs-snapshots", "info", "no snapshot needed", {"volume": volume.id}))
+        logging.info(kayvee.formatLog("ebs-snapshots", "info", "no snapshot needed", {"volume": volume.id, "lastest_snapshot_id": latest_snapshot_id}))
+
+    # Make a backup copy of latest completed snapshot if there is one completed.
+    backup_id = None
+    if latest_complete_snapshot_id is None:
+        logging.info(kayvee.formatLog("ebs-snapshots", "info", "waiting to create backup snapshot until snapshot is complete", {"volume": volume.id}))
+    elif interval == 'hourly' and min_complete_snapshot_delta > 3600:
+        backup_id = _copy_snapshot(backup_client, volume, latest_complete_snapshot_id, name)
+    elif interval == 'daily' and min_complete_snapshot_delta > 3600*24:
+        backup_id = _copy_snapshot(backup_client, volume, latest_complete_snapshot_id, name)
+    elif interval == 'weekly' and min_complete_snapshot_delta > 3600*24*7:
+        backup_id = _copy_snapshot(backup_client, volume, latest_complete_snapshot_id, name)
+    elif interval == 'monthly' and min_complete_snapshot_delta > 3600*24*30:
+        backup_id = _copy_snapshot(backup_client, volume, latest_complete_snapshot_id, name)
+    elif interval == 'yearly' and min_complete_snapshot_delta > 3600*24*365:
+        backup_id = _copy_snapshot(backup_client, volume, latest_complete_snapshot_id, name)
+    elif (backup_id is None) and (not _has_backup(backup_client, volume)):
```

Comment:
Should this `elif` be it's own if-statement?  I ask because `backup_id` is set to `None` on line 192, so I think this case will always be true if `min_complete_snapshot_delta` is a small value or if `interval` is not equal to 'hourly', 'daily', 'weekly', 'monthly', or 'yearly'.